### DISCUSSION TITLE
Bosch Twinguard in Home Assistant

### DISCUSSION THREAD

**Discussion by KorbinianPei** [Koenkk/zigbee2mqtt#22243] [Devices]
Hello everyone,

**Introduction:**
I have a problem with the Bosch Twinguard smoke detectors in Home Assistant. Bosch usually states that the Bosch controller must be used to operate the smoke detectors. This can then also be integrated into Home Assistant via the Bosch integration. But now I saw that the Bosch Twingurad also appears in the list of supported devices on the homepage 
https://www.zigbee2mqtt.io/devices/8750001213.html
For this reason, I tried to integrate my three Twinguards directly into home assistant using Sonoff Dongle P and Zigbee2MQTT.

**Problem Description:**
At the beginning I tried everything using one Twinguard. I connected this as a test right next to the Sonoff. After a few minutes (possibly 30 minutes or a little longer) the Twinguard values (humidity, temperature, CO2 values) also arrived in home assistant. Everything worked wonderfully. 
Then there was a home assistant update (the April Update 2024) and nothing has worked since then. Now I have already installed the core version 2024.4.3 and the supervisor version 2024.04.0, unfortunately without success.

Currently all Twinguards are successfully connected in Zigbee2MQTT in Home Assistant. The devices are also recognized as Twinguards by the integration. When you connect it also says, “Device connected successfully”. Funnily enough, the siren and the self test can also be controlled using the Home Assistant. However, no measured values are transmitted.
I also noticed the following:
In addition to the Home Assistant update, there was an update to the Zigbee2MQTT page regarding the Twinguard. The device page was updated on April 17, 2024 and the VOC value was added. In the Home assistant integration, the VOC value is now also queried in addition to the other values. The integration has certainly changed something that may be independent of the home assistant.

**Summary:**
Anyway, unfortunately all Twinguards do not provide any data after the Home Assistant Update (April 2024) or after the update of the device page on Zigbee2MQTT (VOC-Value added). Only the siren and the self-tests can be controlled.

If any of you have any advice, please let me know.
Thank you very much for helping.

![Twinguard_Home_Assistant_4](https://github.com/Koenkk/zigbee2mqtt/assets/114978654/66de78dd-e25b-4cfc-b3a3-3348a82388f3)
![Twinguard_Home_Assistant_3](https://github.com/Koenkk/zigbee2mqtt/assets/114978654/bf16a884-0195-4715-b3d2-94efb72e3781)
![Twinguard_Home_Assistant_2](https://github.com/Koenkk/zigbee2mqtt/assets/114978654/2d77f9d1-6b5d-4e78-8bd0-764c05aa9dcd)
![Twinguard_Home_Assistant](https://github.com/Koenkk/zigbee2mqtt/assets/114978654/b15e0631-22cb-4076-ae8d-3450b836b2cb)

**Comment by safakaltun** [comment]
I suspect this might be related to the changes made in https://github.com/Koenkk/zigbee-herdsman-converters/issues/7274 by 
@burmistrzak. I have also noticed the same issue today. None of my Twinguards are working anymore... It looks like z2m is looking for the clusters that @burmistrzak removed during his updates. 

@Koenkk could this be related to persistent information stored in one of the database or cache files? 

I am currently on 1.38.0-dev commit: [fd3d9a3].

**Reply by burmistrzak** [reply]
Thanks for pinging me!
All of our four Twinguards are reporting data, so I can't reproduce this on my end.

But as @safakaltun mentioned, some sort of _caching_ could be an issue.
IIRC, I did re-configure all Twinguard out of an abundance of caution. Have you tried this?

@KorbinianPei What's your Z2M version/commit?

I'm still running 1.38.0-dev commit: [457ed86](https://github.com/Koenkk/zigbee2mqtt/commit/457ed86)

**Comment by KorbinianPei** [comment]
Hey, I run z2m version 1.37.1-1.But so far, my twinguards send data for now. It was a problem with the older z2m versions --Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.

 
  
  
 
 Am 15.06.24, 15:13 schrieb burmistrzak ***@***.***>:

   
  Thanks for pinging me! All of our four Twinguards are reporting data, so I can't reproduce this on my end. 
  But as @safakaltun mentioned, some sort of caching could be an issue. IIRC, I did re-configure all Twinguard out of an abundance of caution. Have you tried this? 
  @KorbinianPei What's your Z2M version/commit? 
  I'm still running 1.38.0-dev commit: 457ed86
